avo-hq/avo
# feature: model to resource mapping

# Description


Fixes an issue where if you have multiple resources assigned to the same model, the wrong one might be used to display that resource.

Example: There is a `User` model, and you have the main `UserResource` but also `AnotherUser` resource. When navigating to `/avo/resources/users` the `AnotherUser` resource will be used because it's first in alphabetical order. Now you can change that using the `model_resource_mapping` option.

```ruby
Avo.configure do |config|
  # This will let Avo know that `UserResource` should be used by default for the `User` model.
  config.model_resource_mapping = {
    'User': 'UserResource'
  }
end
```

May fix https://github.com/avo-hq/avo/issues/1162

This PR also changes the `authorize_association_for` method to use the `field.resource` instead of figuring it out based on the model.

[Prerelease](https://rubygems.org/gems/avo/versions/2.13.5.pre.1)
[Docs](https://docs.avohq.io/2.0/resources.html#create-multiple-resources-for-the-same-model)

# Checklist:




## Manual review steps


1. Visit a post.
1. Ensure the user field displays the user name. This tells us that Avo is using the `UserResource`, not the `TeamUserResource` which would be first in alphabetical order.

Manual reviewer: please leave a comment with output from the test if that's the case.
